### PR TITLE
POLIO-1695 POLIO-1697 POLIO-1698 POLIO-1699 POLIO-1696 POLIO-1700 Vaccine module upload of pdf documents

### PR DESCRIPTION
**Pull request by mathvdh**
Upload PDF files into the vaccine module for each type of file:

- VRF
- Pre Alert
- Form A
- Destruction Report
- Incident report

Related JIRA tickets : POLIO-1695, POLIO-1697, POLIO-1698, POLIO-1699, POLIO-1696, POLIO-1700

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [x] My migrations file are included
- [x] Are there enough tests

## Changes

Add a custom storage manager to point to another public S3 Bucket if `AWS_PUBLIC_STORAGE_BUCKET_NAME` env var is set (otherwise it uses default storage)
Add "document" FileField on relevant models that uses this storage
Modification of the APIs to handle the files and uploads.
Needed a new Parser class to handle Nested Multipart Forms data which is supported by the standard but not by DRF out of the box.
Modify the frontend to handle these documents in all the screens for the relevant items (See above)

## How to test

Either directly via API but you would need to send data in Form Data (Not JSON) (See the code in the tests) or via the UI in all the Vaccine Stock and Supply Chain screens.

## Print screen / video

https://github.com/user-attachments/assets/0c8a7765-b625-4fd2-abe5-1dd615a0010a

### CODE REVIEW

File: testdata/test_pdf.pdf
Comment (quang-le) [inline testdata/test_pdf.pdf:1]: Maybe gitignore this?

File: plugins/polio/js/src/domains/VaccineModule/SupplyChain/hooks/api/vrf.tsx [hunk 1]
```diff
@@ -190,17 +193,139 @@ const getRoundsForApi = (
...
+            if (key === 'files' && Array.isArray(value) && value.length > 0) {
+                formData.append('document', value[0]); // Use 'document' key
+            } else if (Array.isArray(value)) {
+                value.forEach((blob, index) => {
+                    formData.append(`${key}[${index}]`, blob);
+                });
+            } else {
+                formData.append(key, value);
+            }
+        });
```
Comment (quang-le) [inline plugins/polio/js/src/domains/VaccineModule/SupplyChain/hooks/api/vrf.tsx]: This can probably be extracted since you repeat the exact same logic in both `patch2` and `post2`.
Not blocking

File: plugins/polio/js/src/domains/VaccineModule/SupplyChain/hooks/api/vrf.tsx [hunk 2]
```diff
@@ -190,17 +193,139 @@ const getRoundsForApi = (
...
+    return iasoFetch(url, init).then(response => response.json());
+};
+
 export const saveVrf = (
     vrf: Optional<Partial<VRFFormData>>,
 ): Promise<any>[] => {
-    const payload: Partial<VRF> = {
-        ...vrf,
-        rounds: getRoundsForApi(vrf?.rounds),
+    const filteredParams = vrf
```
Comment (quang-le) [inline plugins/polio/js/src/domains/VaccineModule/SupplyChain/hooks/api/vrf.tsx:248]: nitpick: `filteredParams` sounds like you're preparing query params for a GET request. Not blocking

File: plugins/polio/js/src/domains/VaccineModule/SupplyChain/Details/VaccineRequestForm/VaccineRequestForm.tsx [hunk 3]
```diff
@@ -15,9 +15,20 @@ import {
...
+export const accept: Accept = {
+    'application/pdf': ['.pdf'],
+};
+
+export const processErrorDocsBase = (err_docs) => {
+    if (!err_docs) return [];
+    if (!Array.isArray(err_docs)) return [err_docs]; 
+    else return err_docs;
+};
+
```
Comment (quang-le) [inline plugins/polio/js/src/domains/VaccineModule/SupplyChain/Details/VaccineRequestForm/VaccineRequestForm.tsx]: I would put this somewhere else, like in utils or something

File: testdata/test_pdf.pdf
Comment (mathvdh) [reply testdata/test_pdf.pdf:1]: I need it for the tests ...

File: plugins/polio/js/src/domains/VaccineModule/SupplyChain/Details/VaccineRequestForm/VaccineRequestForm.tsx (hunk 3 above)
Comment (mathvdh) [reply plugins/polio/js/src/domains/VaccineModule/SupplyChain/Details/VaccineRequestForm/VaccineRequestForm.tsx]: Done

File: plugins/polio/js/src/domains/VaccineModule/SupplyChain/hooks/api/vrf.tsx (hunk 1 above)
Comment (mathvdh) [reply plugins/polio/js/src/domains/VaccineModule/SupplyChain/hooks/api/vrf.tsx]: Done

File: plugins/polio/api/vaccines/supply_chain.py [hunk 4]
```diff
@@ -301,10 +310,25 @@ class Meta:
...
 
+    def to_internal_value(self, data):
+        # Manually invoke validate_rounds if 'rounds' is a string
+        if "rounds" in data and isinstance(data["rounds"], str):
+            try:
+                rounds = parse_comma_separated_numeric_values(data["rounds"], "rounds")
+                data["rounds"] = [{"number": num} for num in rounds]
+            except Exception as e:
+                raise serializers.ValidationError(f"Invalid rounds data: {e}")
+        return super().to_internal_value(data)
```
Comment (bramj) [inline plugins/polio/api/vaccines/supply_chain.py:326]: This looks unrelated to the document upload feature, right?

**Review by bramj — APPROVED** [review]
Tested locally and working! Nothing to comment on the backend code, looks good to me, nice job :+1: 
I only wonder where we are able to see the uploaded documents in the frontend? But maybe that's not important here.

File: plugins/polio/api/vaccines/supply_chain.py (hunk 4 above)
Comment (mathvdh) [reply plugins/polio/api/vaccines/supply_chain.py:326]: It's only related because we changed from JSON to FormData to send the data to the server @bramj and so the old way of sending the rounds was not working
